scripts/install_backend_python_libs.py

```diff
@@ -0,0 +1,406 @@
+            shutil.rmtree(common.THIRD_PARTY_PYTHON_LIBS_DIR)
+        _reinstall_all_dependencies()
+        return
+
+    for library_name, versions in mismatches.items():
+        requirements_version = (
+            pkg_resources.parse_version(versions[0]) if versions[0] else None)
+        directory_version = (
+            pkg_resources.parse_version(versions[1]) if versions[1] else None)
+
+        # The library is installed in the directory but is not listed in
+        # requirements.
+        # We don't have functionality to remove a library cleanly, and if we
+        # ignore the library, this might cause issues when pushing the branch to
+        # develop as there might be possible hidden use cases of a deleted
+        # library that the developer did not catch. The only way to enforce the
+        # removal of a library is to clean out the folder and reinstall
+        # everything from scratch.
+        if not requirements_version:
+            if os.path.isdir(common.THIRD_PARTY_PYTHON_LIBS_DIR):
+                shutil.rmtree(common.THIRD_PARTY_PYTHON_LIBS_DIR)
+            _reinstall_all_dependencies()
+            return
+
+        # The library listed in 'requirements.txt' is not in the
+        # 'third_party/python_libs' directory.
+        if not directory_version:
+            _install_library(
+                library_name,
+                python_utils.convert_to_bytes(requirements_version))
+        # The currently installed library version is not equal to the required
+        # 'requirements.txt' version.
+        elif requirements_version != directory_version:
+            _install_library(
+                library_name,
+                python_utils.convert_to_bytes(requirements_version))
+            _remove_metadata(
+                library_name, python_utils.convert_to_bytes(directory_version))
+
+
+def _install_library(library_name, version_string):
+    """Installs a library with a certain version to the
+    'third_party/python_libs' folder.
+
+    Args:
+        library_name: str. Name of the library to install.
+        version_string: str. Stringified version of the library to install.
+    """
+    subprocess.check_call([
+        'pip', 'install', '--target',
+        common.THIRD_PARTY_PYTHON_LIBS_DIR,
+        '--no-dependencies',
+        '%s==%s' % (
+            library_name,
+            version_string),
+        '--upgrade'
+    ])
+
+
+def _reinstall_all_dependencies():
+    """Reinstalls all of the libraries detailed in the compiled
+    'requirements.txt' file to the 'third_party/python_libs' folder.
+    """
+    subprocess.check_call([
+        'pip', 'install', '--target',
+        common.THIRD_PARTY_PYTHON_LIBS_DIR,
+        '--no-dependencies', '-r',
+        common.COMPILED_REQUIREMENTS_FILE_PATH,
+        '--upgrade'
+    ])
+
+
+def _get_possible_metadata_directory_names(library_name, version_string):
+    """Possible metadata directory names for python libraries installed using
```

Comment:
This is missing a verb (generally methods have verb or verb phrases to start their documentation because it describes what the function is doing). Perhaps: ``Returns possible...``?

Reply:
Fixed.